### ISSUE TITLE
 Dataset Selection for Post-Training Quantization (PTQ) 

### ISSUE THREAD

**Issue by zhangxu223** [intel/neural-compressor#1951]
I have a question regarding the usage of datasets for Post-Training Quantization (PTQ) when using the Neural Compressor.

1. When performing static PTQ, should the val_loader be selected from the validation dataset or the test dataset?

2. For the calibration dataset used during PTQ, should the entire dataset be used, or is it sufficient to select only a subset of the data? If a subset is sufficient, what portion of the data is recommended?

**Comment by zhangxu223** [comment]
My specialty is speech separation, and when I look at the speech recognition examples, both the validation set and the calibration data set use the test data set（test-clean）.  https://github.com/intel/neural-compressor/tree/master/examples/pytorch/speech_recognition/torchaudio_models/quantization/ptq_static/fx   But other examples seem to use validation datasets as calibration datasets and val_loader, is it because of the different tasks that the datasets are selected differently?

**Comment by changwangss** [comment]
> I have a question regarding the usage of datasets for Post-Training Quantization (PTQ) when using the Neural Compressor.
> 
> 1. When performing static PTQ, should the val_loader be selected from the validation dataset or the test dataset?
> 2. For the calibration dataset used during PTQ, should the entire dataset be used, or is it sufficient to select only a subset of the data? If a subset is sufficient, what portion of the data is recommended?

1. both is ok,  you 'd better use validation dataset if we use test dataset as final results, vice versa。
2. You don’t need to use the entire dataset, just a portion of it, usually 100-200 iterations.

**Comment by changwangss** [comment]
> My specialty is speech separation, and when I look at the speech recognition examples, both the validation set and the calibration data set use the test data set（test-clean）.
Because this dataset does not have a validation part，under this condition, only part of the test set can be used for calibration.

**Comment by zhangxu223** [comment]
> > My specialty is speech separation, and when I look at the speech recognition examples, both the validation set and the calibration data set use the test data set（test-clean）.
> 
> Because this dataset does not have a validation part，under this condition, only part of the test set can be used for calibration.

Thank you very much for your answer. But I notice that this dataset has a validation part,  Such as https://github.com/intel/neural-compressor/blob/master/examples/pytorch/speech_recognition/torchaudio_models/quantization/ptq_static/fx/run_asr.py The following code in Line27：parser.add_argument('--url', choices=['dev-clean', 'dev-other', 'test-clean', 'test-other', 
                                      'train-clean-100', 'train-clean-360', 'train-other-500'],
                    default='test-clean',
                    help='the dataset form')   

Is 'dev-clean' the validation portion of the dataset?

**Comment by zhangxu223** [comment]
> > I have a question regarding the usage of datasets for Post-Training Quantization (PTQ) when using the Neural Compressor.
> > 
> > 1. When performing static PTQ, should the val_loader be selected from the validation dataset or the test dataset?
> > 2. For the calibration dataset used during PTQ, should the entire dataset be used, or is it sufficient to select only a subset of the data? If a subset is sufficient, what portion of the data is recommended?
> 
> 1. both is ok,  you 'd better use validation dataset if we use test dataset as final results, vice versa。
> 2. You don’t need to use the entire dataset, just a portion of it, usually 100-200 samples. it is also depended on how many samples you have.

Thank you very much. I understand.
